base/loading.jl

```diff
@@ -2312,10 +2312,14 @@ function include_package_for_output(pkg::PkgId, input::String, depot_path::Vecto
     catch ex
         precompilableerror(ex) || rethrow()
         @debug "Aborting `create_expr_cache'" exception=(ErrorException("Declaration of __precompile__(false) not allowed"), catch_backtrace())
-        exit(125) # we define status = 125 means PrecompileableError
+        exit(125) # we define status = 125 means PrecompilableError
     finally
         Core.Compiler.track_newly_inferred.x = false
     end
+
+    # restore to initial state to make precompilation result relocatable
```

Comment:
Is there a GitHub issue that we could link in the comment here? To provide more explanation about why this is necessary?

Reply:
There is this comment https://github.com/JuliaLang/julia/pull/49866#issuecomment-1765556099 from #49866 which has a very long thread.
And then there was a discussion in Pkg.jl https://github.com/JuliaLang/Pkg.jl/pull/3668#issuecomment-1766719006
But there is no separate issue.

Reply:
Alright, in that case, I think it's fine to not link anything in the comment.